### DISCUSSION TITLE
How to use special prompts with Koboldcpp?

### DISCUSSION THREAD

**Discussion by FlyToha** [LostRuins/koboldcpp#1089] [Q&A] [answered]
Hi! I have a noobie question: what the correct way to using GGUF version of many models with Koboldcpp? For example in the description for some models i can see suggested system prompt and format line. Where exactly i need copy/paste it inside the Koboldcpp? There is few fields like a Memory, Author's note, Author's note templtate and World Info with key words. So can someone tell me, how exactly i need to fill these fields to get correct working any gguf model with special prompts in descriptions?
I've tried to use Codegeex4 without any special prompting, but it answered me with just infinite spam slash symbols. Many other works fine, but i think without special prompting i got not exactly result which a model's author build in it.
![1](https://github.com/user-attachments/assets/afa85a55-8b17-4511-b767-06e672dc73c0)

**Comment by jabberjabberjabber** [comment, marked as answer]
For the first three of those, we will go 1,2,3 for each system tag, user tag, and assistant tag
```
<|im_start|>system
or: 
<s>[INST] 
^note space after [INST]
or:
<|system|>\n

<|im_end|>\n<|im_start|>user\n
or:
  [/INST] </s>[INST] 
^note space before and after [/INST] and after [INST]
or:
\n<|user|>\n

<|im_end|>\n<|im_start|>assistant\n
or:
  [/INST]
^note space before [/INST]
or:
\n<|assistant|>\n
```

**Comment by iblamesupreme** [comment]
@LostRuins You can't just select an answer as _the_ answer, when it only answers a tiny fraction of the entire post. This infuriates me so much right now &ndash; it's 2:34 in the morning, I just downloaded a model I wanted to try for a while, and now it's spitting gibberish at me. Please, fix this.

**Reply by iblamesupreme** [reply]
I found what I was looking for: Settings (top of screen) > Format Tab > System prompt. Good luck, everybody else! Let's see if my comment gets marked too, lol

**Reply by LostRuins** [reply]
Yeah, the system prompt is a big text box that you can paste whatever formatting you like. Though if the model is spitting gibberish its very likely a model issue.

**Reply by iblamesupreme** [reply]
> Yeah, the system prompt is a big text box that you can paste whatever formatting you like. Though if the model is spitting gibberish its very likely a model issue.

Turns out it was.

**Reply by iblamesupreme** [reply]
Also, with "fix this" I meant this thread. The answer you marked as correct is incomplete and doesn't answer all of OP's questions.

**Reply by LostRuins** [reply]
System prompt, author's note, world info and memory all achieve the same thing - they just inject text into the story/prompt at different places. 

System prompt - Always added at the very beginning in system tags
Memory - Added after system prompt, raw format (add your own tags if desired)
Authors Note - Added a few lines before the end of the prompt for greater impact
World Info - Added only if specific text matches the search key (configurable)

At the end of the day, text is text and works the same. It's just how you want to use it.

**Reply by gustrd** [reply]
My 2 cents: what is being called here "special prompts" seems to be "prompt templates". 

Sadly there is very little good material online explaining the concept.
